open_lm/main.py

```diff
@@ -100,18 +100,20 @@ def get_state_dict(name):
     return sd
 
 
-def load_model(args, model):
+def load_model(args, model, different_seed=False):
     checkpoint = pt_load(args.resume, map_location="cpu")
     if "epoch" in checkpoint:
-        if "shard_shuffle_seed" in checkpoint:
+        if not different_seed and "shard_shuffle_seed" in checkpoint:
             pretrained_seed = checkpoint["shard_shuffle_seed"]
             assert (
                 pretrained_seed == args.seed
             ), f"This checkpoint was trained with a random seed of {pretrained_seed}. Since this seed affects shard shuffling, resuming training must use the same seed."
         else:
-            logging.info(
-                "Resuming a checkpoint that does not have a seed saved. This means that the shards were not shuffled, so they will remain unshuffled."
-            )
+            if "shard_shuffle_seed" in checkpoint:
+                message = "Resuming a checkpoint that does not have a seed saved. This means that the shards were not shuffled, so they will remain unshuffled."
+            else:
+                message = "Resuming a checkpoint without checking that the seed match. This means that training might not be reproducible."
+            logging.info(message)
             pretrained_seed = None
```

Comment:
I think the structure should be better as e.g. if different_seed then print a message that we ignore seed checking
and if not different_seed then do the exact same thing as before. I think this makes the messages more consistent.